recipes/xmlsec/all/conanfile.py

```diff
@@ -0,0 +1,213 @@
+            "--enable-apps-crypto-dl={}".format(yes_no(False)),
+            "--with-libxslt={}".format(yes_no(self.options.with_xslt)),
+            "--with-openssl={}".format(tools.unix_path(self.deps_cpp_info["openssl"].rootpath) if self.options.with_openssl else "no"),
+            "--enable-mscrypto={}".format(yes_no(False)),   # Built on mingw
+            "--enable-mscng={}".format(yes_no(False)),      # Build on mingw
+            "--enable-docs={}".format(yes_no(False)),
+            "--enable-mans={}".format(yes_no(False)),
+            "--enable-shared={}".format(yes_no(self.options.shared)),
+            "--enable-static={}".format(yes_no(not self.options.shared)),
+        ]
+        self._autotools.libs = []
+        self._autotools.configure(args=configure_args, configure_dir=self._source_subfolder)
+        return self._autotools
+
+    def build(self):
+        if self._is_msvc:
+            self._build_msvc()
+        else:
+            # FIXME: conan is missing a feature to change the join character on lists of environment variables (fix for mingw on Windows)
+            # e.g. self.env_info.AUTOMAKE_CONAN_INCLUDES.joiner = ":"
+            with tools.environment_append({"AUTOMAKE_CONAN_INCLUDES": tools.get_env("AUTOMAKE_CONAN_INCLUDES", "").replace(";", ":")}):
+                with tools.chdir(self._source_subfolder):
+                    self.run("autoreconf -fiv", run_environment=True, win_bash=tools.os_info.is_windows)
+            autotools = self._configure_autotools()
+            autotools.make()
+
+    def package(self):
+        self.copy("Copyright", src=self._source_subfolder, dst="licenses")
+
+        if self._is_msvc:
+            self._package_msvc()
+            if not self.options.shared:
+                tools.remove_files_by_mask(os.path.join(self.package_folder, "bin"), "*.dll")
+            tools.remove_files_by_mask(os.path.join(self.package_folder, "bin"), "*.pdb")
+            os.unlink(os.path.join(self.package_folder, "lib", "libxmlsec-openssl_a.lib" if self.options.shared else "libxmlsec-openssl.lib"))
+            os.unlink(os.path.join(self.package_folder, "lib", "libxmlsec_a.lib" if self.options.shared else "libxmlsec.lib"))
+        else:
+            autotools = self._configure_autotools()
+            autotools.install()
+            tools.rmdir(os.path.join(self.package_folder, "share"))
+            tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.la")
+            tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+            os.remove(os.path.join(self.package_folder, "lib", "xmlsec1Conf.sh"))
+
+    def package_info(self):
+        prefix = "lib" if self._is_msvc else ""
+        infix = "" if self._is_msvc else str(tools.Version(self.version).major)
+        suffix = "_a" if self._is_msvc and not self.options.shared else ""
+
+        get_libname = lambda libname: prefix + "xmlsec" + infix + (("-" + libname) if libname else "") + suffix
```

Comment:
This is the most complicated thing I've seen... is this the meaning of life ?!?!

Reply:
The meaning of life would just be `get_libname = lambda libname: 42` :smile: 

It's written like a lambda because it must calculate a lib name for every component (2 as of now).
When some crypto engine is added, this lambda can be re-used.